src/Model/BSONDocument.php

```
@@ -52,10 +52,10 @@ public function __clone()
      * by default.
      *
      * @see https://php.net/arrayobject.construct
-     * @param array<string, mixed> $input
+     * @param object|array<string, mixed> $input
      * @psalm-param class-string<ArrayIterator<string,mixed>>|class-string<ArrayObject<string,mixed>> $iteratorClass
      */
-    public function __construct(array $input = [], int $flags = ArrayObject::ARRAY_AS_PROPS, string $iteratorClass = ArrayIterator::class)
+    public function __construct(array|object $input = [], int $flags = ArrayObject::ARRAY_AS_PROPS, string $iteratorClass = ArrayIterator::class)
```

I'll have to dig this error.
```
ERROR: PossiblyInvalidArgument
at /Users/jerome/Develop/mongo-php-library/src/Model/BSONDocument.php:60:29
Argument 1 of ArrayObject::__construct expects (array<TKey:ArrayObject as array-key, TValue:ArrayObject as mixed>)|null|object, but possibly different type array<string, mixed>|object provided (see https://psalm.dev/092)
        parent::__construct($input, $flags, $iteratorClass);

```

---

This may be related: https://github.com/vimeo/psalm/issues/9082

---

Thanks for checking. So I think we can ignore this error.

---

This psalm issue was solved by moving to `array|stdClass`.